pkg/provider/pulumi/pulumi.go

```diff
@@ -0,0 +1,90 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package pulumi
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"time"
+
+	esc "github.com/pulumi/esc/cmd/esc/cli/client"
+	corev1 "k8s.io/api/core/v1"
+
+	esv1beta1 "github.com/external-secrets/external-secrets/apis/externalsecrets/v1beta1"
+)
+
+type client struct {
+	escClient    esc.Client
+	environment  string
+	organization string
+	duration     time.Duration
+}
+
+var _ esv1beta1.SecretsClient = &client{}
+
+func (c *client) GetSecret(ctx context.Context, ref esv1beta1.ExternalSecretDataRemoteRef) ([]byte, error) {
+	x, _, err := c.escClient.OpenEnvironment(ctx, c.organization, c.environment, c.duration)
+	if err != nil {
+		return nil, err
+	}
+	value, err := c.escClient.GetOpenProperty(ctx, c.organization, c.environment, x, ref.Key)
+	if err != nil {
+		return nil, err
+	}
+	return []byte(value.ToString(false)), nil
+}
+
+func (c *client) PushSecret(_ context.Context, _ *corev1.Secret, _ esv1beta1.PushSecretData) error {
+	return errors.New("pushing secrets is currently not supported by Pulumi")
+}
+
+func (c *client) DeleteSecret(_ context.Context, _ esv1beta1.PushSecretRemoteRef) error {
+	return errors.New("deleting secrets is currently not supported by Pulumi")
+}
+
+func (c *client) Validate() (esv1beta1.ValidationResult, error) {
+	return esv1beta1.ValidationResultReady, nil
+}
+
+func (c *client) GetSecretMap(ctx context.Context, ref esv1beta1.ExternalSecretDataRemoteRef) (map[string][]byte, error) {
+	data, err := c.GetSecret(ctx, ref)
+	if err != nil {
+		return nil, fmt.Errorf("error getting secret %s: %w", ref.Key, err)
+	}
+
+	// Maps the json data to a string:string map
+	kv := make(map[string]string)
```

Comment:
This is a little bit too limited, users could define a `string|number|object|array|bool|null` as value. un-marshaling would fail below.

Maybe this could be an approach, though i'm open for suggestions :)
```go
maybeMap := value.ToJSON(false)
myMap, ok := maybeMap.(map[string]any); 
if !ok {
  // not a map, return err
}

output := make(map[string][]byte)
for key := range myMap {
  output[key] = utils.GetByteValue(myMap[key])
}
return output
```